TESTS/mbed_drivers/lp_timer/main.cpp

```diff
@@ -0,0 +1,410 @@
+ * Then timer read functions returns only the the second
+ * measured time.
+ */
+void test_lptimer_reset()
+{
+    LowPowerTimer lp_timer;
+
+    /* First measure 10 ms delay. */
+    lp_timer.start();
+
+    /* Wait 10 ms. */
+    wait_ms(10);
+
+    /* Stop the timer. */
+    lp_timer.stop();
+
+    /* Check results - totally 10 ms elapsed. */
+    TEST_ASSERT_FLOAT_WITHIN(DELTA_S, 0.010f, lp_timer.read());
+    TEST_ASSERT_INT32_WITHIN(DELTA_MS, 10, lp_timer.read_ms());
+    TEST_ASSERT_INT32_WITHIN(DELTA_US, 10000, lp_timer.read_us());
+    TEST_ASSERT_UINT64_WITHIN(DELTA_US, 10000, lp_timer.read_high_resolution_us());
+
+    /* Reset the timer - previous measured time should be lost now. */
+    lp_timer.reset();
+
+    /* Now measure 20 ms delay. */
+    lp_timer.start();
+
+    /* Wait 20 ms. */
+    wait_ms(20);
+
+    /* Stop the timer. */
+    lp_timer.stop();
+
+    /* Check results - 20 ms elapsed since the reset. */
+    TEST_ASSERT_FLOAT_WITHIN(DELTA_S, 0.020f, lp_timer.read());
+    TEST_ASSERT_INT32_WITHIN(DELTA_MS, 20, lp_timer.read_ms());
+    TEST_ASSERT_INT32_WITHIN(DELTA_US, 20000, lp_timer.read_us());
+    TEST_ASSERT_UINT64_WITHIN(DELTA_US, 20000, lp_timer.read_high_resolution_us());
+}
+
+/* This test verifies if calling start() for already
+ * started low power timer does nothing.
+ *
+ * Timer is already started.
+ * When timer is started again.
+ * Then second start operation is ignored.
+ */
+void test_lptimer_start_started_timer()
+{
+    LowPowerTimer lp_timer;
+
+    /* Start the timer. */
+    lp_timer.start();
+
+    /* Wait 10 ms. */
+    wait_ms(10);
+
+    /* Now start timer again. */
+    lp_timer.start();
+
+    /* Wait 20 ms. */
+    wait_ms(20);
+
+    /* Stop the timer. */
+    lp_timer.stop();
+
+    /* Check results - 30 ms have elapsed since the first start. */
+    TEST_ASSERT_FLOAT_WITHIN(2 * DELTA_S, 0.030f, lp_timer.read());
+    TEST_ASSERT_INT32_WITHIN(DELTA_MS, 30, lp_timer.read_ms());
+    TEST_ASSERT_INT32_WITHIN(2 * DELTA_US, 30000, lp_timer.read_us());
+    TEST_ASSERT_UINT64_WITHIN(2 * DELTA_US, 30000, lp_timer.read_high_resolution_us());
+}
+
+/* This test verifies low power timer float operator.
+ *
+ * Timer is created and a time period time is counted.
+ * When timer object is casted on float type.
+ * Then counted type in seconds is returned by means of
+ * read() function.
+ */
+void test_lptimer_float_operator()
+{
+    LowPowerTimer lp_timer;
+
+    /* Start the timer. */
+    lp_timer.start();
+
+    /* Wait 10 ms. */
+    wait_ms(10);
+
+    /* Stop the timer. */
+    lp_timer.stop();
+
+    /* Check result - 10 ms elapsed. */
+    TEST_ASSERT_FLOAT_WITHIN(DELTA_S, 0.010f, (float )(lp_timer));
+}
+
+/* This test verifies if time counted by the low power timer is
+ * valid.
+ *
+ * Timer is created.
+ * When timer is used to measure 100us/1ms/10ms/1s/10s
+ * delays.
+ * Then the results are valid (within acceptable range).
+ */
+void test_lptimer_time_measurement()
```

Comment:
Can we make this function an template and add multiple test cases instead of duplicate the code.

Reply:
To be corrected.

Reply:
fixed.